pkg/wallet/wallet.go

```diff
@@ -81,6 +81,38 @@ func (w *BaseWallet) CreateSigningKey() (string, error) {
 	return base58Pub, nil
 }
 
+// ConvertToEncryptionKey converts an ed25519 keypair present in the wallet,
+// persists the resulting keypair, and returns the result public key.
+func (w *BaseWallet) ConvertToEncryptionKey(key string) (string, error) {
+	signPub := base58.Decode(key)
+	encPub, err := cryptoutil.PublicEd25519toCurve25519(signPub)
+	if err != nil {
+		return "", err
+	}
+	encPubB58 := base58.Encode(encPub)
+
+	kp, err := w.getKey(key)
+	if err != nil {
+		return "", err
+	}
+	encPriv, err := cryptoutil.SecretEd25519toCurve25519(kp.Priv)
+	if err != nil {
+		return "", err
+	}
+	kpEnc := cryptoutil.KeyPair{Priv: encPriv, Pub: encPub}
+	err = w.persistKey(encPubB58, &kpEnc)
+	if err != nil {
+		return "", err
+	}
+
+	return encPubB58, nil
+}
+
+// GetBox returns a Box which provides crypto box encryption which can use this wallet's secret keys
```

Comment:
It still seems a bit odd that we have box methods (rather than just normal crypto primitives exposed). I haven’t thought of a good alternative though. We might need to rethink in the future.

Reply:
@Moopli I wonder if it would make sense to have an internal key package that exposes private material only to framework packages.

The wallet could return a handle to the key (pointer) and the box package could take that handle in its constructor... but the accessors would be unaccessible to outside packages. This way box functionality can be extracted from the wallet package.

Reply:
created #511

Reply:
@troyronda that would increase risks of leaking the secrets - see [Managing Secrets](https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0050-wallets/README.md#managing-secrets)

Have other solutions been explored such as the wallet returning a `Signer` or `Encrypter`?